### DISCUSSION TITLE
grammar - entry/exit points

### DISCUSSION THREAD

**Discussion by adamfk** [StateSmith/StateSmith#210] [Ideas]
There's some [discussion](https://github.com/StateSmith/StateSmith/issues/204#issuecomment-1563990673) about the exit point grammar.

I currently prefer having the `via` stuff at the end as it works well for both entry and exit points. It also seems less syntactically complex to me although that is subjective.

If we move the via stuff to the front, does that also affect entry points? The current syntax `1. EV1 via entry 1` reads better to me than `via entry 1 1. EV1`.

![image](https://github.com/StateSmith/StateSmith/assets/274012/7d640c4a-3ad3-4969-a4e8-0fdee84e1d56)

New lines can be used to help visually and syntactically separate user code from via entry/exit.

![image](https://github.com/StateSmith/StateSmith/assets/274012/282e1dee-ee5c-4106-ab29-c0d6a919c9ae)

I think the existing grammar is OK for now. It definitely could be improved, but I would rather see us put effort into supporting entry exit points [visually instead](https://github.com/StateSmith/StateSmith/issues/209).

**Comment by iamjuing** [comment]
If there is an easy way to represent exit/entry points visually in draw.io (in some flexible way), it should be the top priority for sure.
